server/routes/referral.routes.js

```diff
@@ -0,0 +1,188 @@
+import express from 'express';
+import { authenticate, schemaValidation, isAdmin, hasValidObjectId } from '../helpers/middleware';
+import {
+  updateReferralSchema,
+  emailReferralSchema,
+  sendReferralSchema,
+} from '../schemas/referral.schema';
+import ReferralController from '../controllers/referral.controllers';
+
+const router = express.Router();
+
+/**
+ * @swagger
+ * /referral/:
+ *   get:
+ *     tags:
+ *       - Referral
+ *     description: Get all owned referrals
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Referral'
+ *      description: Get all owned referrals
+ *     responses:
+ *      '200':
+ *        description: returns object of referrals
+ *      '500':
+ *       description: Internal server error
+ */
+router.get('/', authenticate, ReferralController.getUserReferrals);
+
+/**
+ * @swagger
+ * /referral/all/:
+ *   get:
+ *     tags:
+ *       - Referral
+ *     description: Get all referrals
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Referral'
+ *      description: Get all referrals
+ *     responses:
+ *      '200':
+ *        description: returns object of referrals
+ *      '500':
+ *       description: Internal server error
+ */
+router.get('/all', authenticate, isAdmin, ReferralController.getAllReferrals);
+
+/**
+ * @swagger
+ * /referral/rewarded:
+ *   put:
+ *     tags:
+ *       - Referral
+ *     description: Approves a specific referral
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Referral'
+ *      description: Approves a specific referral
+ *     responses:
+ *      '200':
+ *        description: Referral approved
+ *      '400':
+ *        description: Error approving referral
+ *      '500':
+ *       description: Internal server error
+ */
+router.put(
+  '/rewarded',
+  authenticate,
+  isAdmin,
+  schemaValidation(updateReferralSchema),
+  ReferralController.updateReferralToRewarded,
+);
+
+/**
+ * @swagger
+ * /referral/email:
+ *   post:
+ *     tags:
+ *       - Referral
+ *     description: Get a referral by email
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Referral'
+ *      description: Get a referral by email
+ *     responses:
+ *      '200':
+ *        description: Referral found
+ *      '404':
+ *        description: Referral not found
+ *      '500':
+ *       description: Internal server error
+ */
+router.post('/email', schemaValidation(emailReferralSchema), ReferralController.getByEmail);
```

Comment:
Do you recall the use case for this? I am sorry, I can't think of the reason why we would want to get a referral by email (this can be problematic as we allow multiple email invites from different registered member).

If you can remember, can you state the reason here else can we remove it the enpoint please.

Reply:
Also, if we are getting by email, the method should be get not post.

```suggestion
router.get('/email', schemaValidation(emailReferralSchema), ReferralController.getByEmail);
```

Reply:
I can't recall either, but i think you asked for it for some reason, i'd could remove it tho

Reply:
Please let's remove it. We can also come to this PR to copy the code if we later need it.